catroid/src/main/java/org/catrobat/catroid/content/bricks/SetPenColorBrick.java

```diff
@@ -154,29 +154,54 @@ public void showFormulaEditorToEditFormula(View view) {
 	}
 
 	private boolean areAllBrickFieldsNumbers() {
-		return (getFormulaWithBrickField(BrickField.PHIRO_LIGHT_RED).getRoot().getElementType() == FormulaElement.ElementType.NUMBER)
-				&& (getFormulaWithBrickField(BrickField.PHIRO_LIGHT_GREEN).getRoot().getElementType() == FormulaElement.ElementType.NUMBER)
-				&& (getFormulaWithBrickField(BrickField.PHIRO_LIGHT_BLUE).getRoot().getElementType() == FormulaElement.ElementType.NUMBER);
+		return (getFormulaWithBrickField(BrickField.PEN_COLOR_RED).getRoot().getElementType() == FormulaElement.ElementType.NUMBER)
+				&& (getFormulaWithBrickField(BrickField.PEN_COLOR_GREEN).getRoot().getElementType() == FormulaElement.ElementType.NUMBER)
+				&& (getFormulaWithBrickField(BrickField.PEN_COLOR_BLUE).getRoot().getElementType() == FormulaElement.ElementType.NUMBER);
 	}
 
 	private BrickField getClickedBrickField(View view) {
 		switch (view.getId()) {
 			case R.id.brick_set_pen_color_action_green_edit_text:
-				return BrickField.PHIRO_LIGHT_GREEN;
+				return BrickField.PEN_COLOR_GREEN;
 			case R.id.brick_set_pen_color_action_blue_edit_text:
-				return BrickField.PHIRO_LIGHT_BLUE;
+				return BrickField.PEN_COLOR_BLUE;
 			case R.id.brick_set_pen_color_action_red_edit_text:
 			default:
-				return BrickField.PHIRO_LIGHT_RED;
+				return BrickField.PEN_COLOR_RED;
 		}
 	}
 
 	@Override
 	public List<SequenceAction> addActionToSequence(Sprite sprite, SequenceAction sequence) {
 		sequence.addAction(sprite.getActionFactory().createSetPenColorAction(sprite,
-				getFormulaWithBrickField(BrickField.PHIRO_LIGHT_RED),
-				getFormulaWithBrickField(BrickField.PHIRO_LIGHT_GREEN),
-				getFormulaWithBrickField(BrickField.PHIRO_LIGHT_BLUE)));
+				getFormulaWithBrickField(BrickField.PEN_COLOR_RED),
+				getFormulaWithBrickField(BrickField.PEN_COLOR_GREEN),
+				getFormulaWithBrickField(BrickField.PEN_COLOR_BLUE)));
 		return null;
 	}
+
+	public void correctBrickFieldsFromPhiro() {
+		Formula formulaRed;
+		Formula formulaGreen;
+		Formula formulaBlue;
+
+		try {
+			formulaRed = getFormulaWithBrickField(BrickField.PHIRO_LIGHT_RED);
+			formulaGreen = getFormulaWithBrickField(BrickField.PHIRO_LIGHT_GREEN);
+			formulaBlue = getFormulaWithBrickField(BrickField.PHIRO_LIGHT_BLUE);
+		} catch (IllegalArgumentException ex) {
+			// brick is already correct
```

Comment:
I'm not sure about using exceptions for flow control, especially since the case that the brick is already correct is not really an exception (should happen quite a bit). What do you think?